### PR TITLE
Fix -Werror=conversion and -Wdeprecated-copy issues

### LINKED ISSUE #3754: "-Werror=conversion" error after merging PR #3718 in Linux
Building yuzu on master and mainline in Linux, I get the following errors:
```
In file included from yuzu/src/./video_core/engines/kepler_compute.h:10,
                 from yuzu/src/video_core/renderer_vulkan/vk_rasterizer.cpp:20:
yuzu/src/./common/bit_field.h: In instantiation of ‘constexpr void BitField<Position, Bits, T, EndianTag>::Assign(const T&) [with long unsigned int Position = 0; long unsigned int Bits = 1; T = short unsigned int; EndianTag = KeepTag]’:
yuzu/src/./video_core/renderer_vulkan/fixed_pipeline_state.h:131:51:   required from here
yuzu/src/./common/bit_field.h:183:63: error: conversion from ‘int’ to ‘BitField<0, 1, short unsigned int>::StorageTypeWithEndian’ {aka ‘short unsigned int’} may change value [-Werror=conversion]
  183 |         storage = (static_cast<StorageType>(storage) & ~mask) | FormatValue(value);
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
```
```
In file included from yuzu/src/./video_core/renderer_vulkan/vk_rasterizer.h:20,
                 from yuzu/src/video_core/renderer_vulkan/renderer_vulkan.cpp:31:
yuzu/src/./video_core/renderer_vulkan/fixed_pipeline_state.h: In member function ‘void Vulkan::FixedPipelineState::VertexInput::SetBinding(std::size_t, bool, u32, u32)’:
yuzu/src/./video_core/renderer_vulkan/fixed_pipeline_state.h:132:35: error: conversion from ‘u32’ {aka ‘unsigned int’} to ‘short unsigned int’ may change value [-Werror=conversion]
  132 |             binding.stride.Assign(stride);
      |                                   ^~~~~~
```

The second was obtained by modifying the first error to `storage = (u_int16_t)(static_cast<StorageType>(storage) & ~mask) | (u_int16_t)FormatValue(value);`. If I modify the second to read `binding.stride.Assign((u_int16_t)stride);`, the build will complete, but games softlock immediately, so there is something I don't understand well enough to fix the error correctly.

My system specs are:
Manjaro Linux 20.0
Kernel 5.6.5-10-tkg-pds
Ryzen 7 2700X
gcc (Arch Linux 9.3.0-1) 9.3.0

The error is NOT detected by the Linux:MingW Docker build environment (`-DENABLE_VULKAN=OFF` is a CMake argument in `.ci/scripts/windows/docker.sh`). I do not have the Linux Docker environment setup.

### PR DESCRIPTION
**Pull request by degasus**
Fixes #3754

### CODE REVIEW

**Comment by Morph1984** [comment]
Fixes #3754 ?

**Comment by lat9nq** [comment]
Fixes #3754

File: src/video_core/renderer_vulkan/fixed_pipeline_state.h [hunk 1]
```diff
@@ -129,7 +129,7 @@ struct alignas(32) FixedPipelineState {
             auto& binding = bindings[index];
             binding.raw = 0;
             binding.enabled.Assign(enabled ? 1 : 0);
-            binding.stride.Assign(stride);
+            binding.stride.Assign(static_cast<u16>(stride));
```
Comment (degasus) [inline src/video_core/renderer_vulkan/fixed_pipeline_state.h:132]: @ReinUsesLisp Might you check if this cast is correct? We get a 32 bit value and truncate it to 14 bit.

Comment (ReinUsesLisp) [reply src/video_core/renderer_vulkan/fixed_pipeline_state.h:132]: Yes, it' safe truncating it. The hardware register is an u12:
https://github.com/yuzu-emu/yuzu/blob/8f548266cd0493a537ed7edf56c869bd7420953a/src/video_core/engines/maxwell_3d.h#L1236

I used u14 on `binding.stride` just for padding, although it's not necessary.

**Pull request merged**